lib/json_api/player_deserializer.rb

```diff
@@ -1,13 +1,38 @@
 require 'game/player'
+require 'json_api/errors'
 
 module JsonAPI
   class PlayerDeserializer
-    def deserialize(serialized_players)
-      players = deserialize_players(serialized_players)
+    def deserialize(request_body)
+      raise_error_if_invalid(request_body)
+      deserialize_players(request_body['players'])
     end
 
     private
 
+    def raise_error_if_invalid(request_body)
+      raise DeserializationError.new("a player is missing") if missing_players?(request_body)
+      raise DeserializationError.new("player is invalid") if invalid_players?(request_body['players'])
+    end
+
+    def missing_players?(request_body)
+      !request_body.has_key?('players') ||
+        !request_body['players'].has_key?('player_one') ||
+        !request_body['players'].has_key?('player_two')
+    end
+
+    def invalid_players?(players)
+      invalid = false
+      players.values.each { |player| invalid = true if invalid_player(player) }
```

Comment:
Could you refactor this so it runs through it and finds all cases of invalid players and if count  > 0 then that's the return? So you're not assigning a variable and switching it

Reply:
Yes, I like that suggestion.